core/api/graphql/mutation_create_artifact.graphql

```diff
@@ -1,46 +1,27 @@
-mutation CreateArtifact(
-    $entityName: String!,
-    $projectName: String!,
-    $artifactTypeName: String!,
-    $artifactCollectionName: String!,
-    $runName: String,
-    $digest: String!,
-    $description: String,
-    $aliases: [ArtifactAliasInput!],
-    $metadata: JSONString,
-    $ttlDurationSeconds: Int64,
-    $historyStep: Int64,
-    $distributedID: String,
-    $clientID: ID!,
-    $sequenceClientID: ID!,
-) {
-    createArtifact(input: {
-        entityName: $entityName,
-        projectName: $projectName,
-        artifactTypeName: $artifactTypeName,
-        artifactCollectionName: $artifactCollectionName,
-        runName: $runName,
-        digest: $digest,
-        digestAlgorithm: MANIFEST_MD5,
-        description: $description,
-        aliases: $aliases,
-        metadata: $metadata,
-        ttlDurationSeconds: $ttlDurationSeconds,
-        historyStep: $historyStep,
-        enableDigestDeduplication: true,
-        distributedID: $distributedID,
-        clientID: $clientID,
-        sequenceClientID: $sequenceClientID,
-        }
-    ) {
-        artifact {
-            id
-            state
-            artifactSequence {
-                latestArtifact {
-                    id
-                }
+fragment CreatedArtifact on CreateArtifactPayload {
+    artifact {
+        id
+        state
+        artifactSequence {
+            latestArtifact {
+                id
             }
         }
     }
 }
+
+# These are "required" fields in practice, even if they're formally optional in the input schema
+# @genqlient(for: "CreateArtifactInput.artifactCollectionName", pointer: false)
+# @genqlient(for: "CreateArtifactInput.enableDigestDeduplication", pointer: false)
+# @genqlient(for: "CreateArtifactInput.clientID", pointer: false)
+# @genqlient(for: "CreateArtifactInput.sequenceClientID", pointer: false)
+#
+# CreateArtifactInput.tags added in server version: 0.58
+# @genqlient(for: "CreateArtifactInput.tags", omitempty: true)
+mutation CreateArtifact(
+    $input: CreateArtifactInput!
+) {
+    createArtifact(input: $input) {
+        ...CreatedArtifact
+    }
+}
```

Comment:
context: Since we're already using `genqlient`, this change leans in heavily on using the `@genqlient` directive to ensure empty/null fields are omitted from graphql requests.

Note: this allows more flexibility for accommodating updated schemas, so long as newer fields are omitted when making GQL requests to older server versions (that don't support those fields).   This does mean that the *client code* is more directly responsible for:
- identifying which fields are "unsupported"
- nulling/zeroing/emptying out any "unsupported" fields

For more info: https://github.com/Khan/genqlient/blob/main/docs/genqlient_directive.graphql